pkg/process/metadata/metadata.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package metadata
+
+import "github.com/DataDog/datadog-agent/pkg/process/procutil"
+
+// Extractor is common interface for Process metadata extraction
+type Extractor interface {
+	Extract(p *procutil.Process)
+	Type() string
+}
```

Comment:
I sort of understand the motivation here (eg. standardize things that use process data), but I don't fully get the benefit of "abiding" to this interface (at least for now :sweat_smile: ). For example, what effectively _uses_ (as in, takes as an argument) a `metadata.Extractor`?

Reply:
Hey @p-lambert , maybe you and @wiyu had that conversation, but we are trying to come up with a model that can be adapted to the future component architecture. Would it be helpful if the flow where we call a `ProcessData` => extractors called, then they are just fetched and called directly in a check? What that means effectively:

Today we have something like this:
```
    // Filter out (in-place) connection data associated with docker-proxy
	procs, err := c.probe.ProcessesByPID(time.Now(), false)
    // ...
	dockerproxy.NewFilter(procs).Filter(conns)
```

Intermediary step:
```
    // Ensure process data is up-to-date (just a wrapper around c.probe.ProcessesByPID)
	ProcessData.Fetch()
    // ... 
    // `DockerProxy` implements an `Extractor` attached to `ProcessData`, receives updates from it
	c.dockerFilter.Filter(conns)
```

Final step:
```
    // `DockerProxy` implements an `Extractor` attached to `ProcessData`, receives updates from it as before,
    // but the scheduling of the check ensures data is collected for users of extractors to consume
	c.dockerFilter.Filter(conns)
```

Reply:
@p-lambert actually realizing now your other comment is pretty close to this idea ^:
> Yup, I agree that in that context it would be useful to have an interface because maybe you'll somehow register a metadata.Extractor to a global process check of some sort, and this global thing would execute these Extract methods as callbacks or something like that.